### ISSUE TITLE
Cannot reproduce Envelope results

### ISSUE THREAD

**Issue by AsadJeewa** [LucasAlegre/morl-baselines#176]
I have been trying to reproduce the results of Envelope by matching the hyperparameters from [https://wandb.ai/openrlbenchmark/MORL-Baselines](https://wandb.ai/openrlbenchmark/MORL-Baselines), I have not been seeing any convergence even when running for 100k. 

Please advise.

[https://wandb.ai/openrlbenchmark/MORL-Baselines/runs/zt92ve1k?nw=nwuseraraffin](https://wandb.ai/openrlbenchmark/MORL-Baselines/runs/zt92ve1k?nw=nwuseraraffin)

```
algo:"Envelope"
batch_size:32
buffer_size:1,500,000
clip_grand_norm:1
env_id:"minecart-v0"
epsilon_decay_steps:100,000
eval_freq:1,000
final_homotopy_lambda:1
gamma:0.99
gradient_updates:1
homotopy_decay_steps:null
initial_epsilon:1
initial_homotopy_lambda:0
known_front:null
learning_rate:0.0002
learning_starts:100
net_arch:
0:256
1:256
2:256
3:256
num_eval_episodes_for_front:5
num_eval_weights_for_eval:50
num_eval_weights_for_front:100
num_sample_w:4
per:true
ref_point:
0:-1
1:-1
2:-200
reset_learning_starts:false
reset_num_timesteps:true
seed:null
target_net_update_freq:200
tau:0.1
total_episodes:null
total_timesteps:1,000,000
use_envelope:true
weight:null
```

<img width="898" height="458" alt="Image" src="https://github.com/user-attachments/assets/9cc1541d-d366-4e8e-ba37-cf97a45db398" />

It is also not logging eval metrics as the other wandb graphs are close to 100k. I have tried fiddling with the hyperparameters.

**Comment by LucasAlegre** [comment]
The x-axis should be changed to "global_step" (number of environment interactions) and not "step".

How many seeds have you run?

**Comment by LucasAlegre** [comment]
I do not have any clues, if you indeed used the exact same hyperparameters... You can try to checkout the library at the git state in https://wandb.ai/openrlbenchmark/MORL-Baselines/runs/zt92ve1k?nw=nwuseraraffin and try to see if anything changed in the code.

**Comment by AsadJeewa** [comment]
I was eventually able to get it to work with some tuning and these hyperparam. Issue can be closed.

[env_minecart.json](https://github.com/user-attachments/files/31826513/env_minecart.json)
